internal/driver/driver.go

```diff
@@ -452,33 +455,31 @@ func (d *Driver) RemoveDevice(deviceName string, protocols map[string]models.Pro
 	return nil
 }
 
-// RefreshMultipleDevicePaths runs RefreshSingleDevicePaths for every connected device
-func (d *Driver) RefreshMultipleDevicePaths() {
+// RefreshAllDevicePaths runs RefreshDevicePaths for every connected device
+func (d *Driver) RefreshAllDevicePaths() {
 	d.lc.Debug("Refreshing existing device paths")
 	for _, cd := range d.ds.Devices() {
-		d.RefreshSingleDevicePaths(cd)
+		d.RefreshDevicePaths(cd)
 	}
 }
 
-// RefreshSingleDevicePaths checks whether the existing devices match the connected devices.
+// RefreshDevicePaths checks whether the existing device matches the connected device.
 // If there is a mismatch between them, scan all paths to find the matching device
 // and update the existing device with the correct path.
-func (d *Driver) RefreshSingleDevicePaths(cd models.Device) {
-	d.lc.Debug("Refreshing existing device paths")
+func (d *Driver) RefreshDevicePaths(cd models.Device) {
 	paths, err := d.getPaths(cd.Protocols)
 	if err != nil {
 		d.lc.Errorf("Failed to get paths for device %s", cd.Name)
 	}
 	for _, fdPath := range paths {
-		cn, sn, err := getUSBDeviceIdInfo(fdPath)
+		_, sn, err := getUSBDeviceIdInfo(fdPath)
 		if err != nil {
 			d.lc.Errorf("failed to get the serial number of device %s, error: %s", cd.Name, err.Error())
 		}
-		// If the card name or serial number is different, it means that the path of the device has changed.
-		if cn != cd.Protocols[UsbProtocol][CardName] || sn != cd.Protocols[UsbProtocol][SerialNumber] || !d.isVideoCaptureDevice(fdPath) {
+		// If the serial number is different, it means that the path of the device has changed.
+		if sn != cd.Protocols[UsbProtocol][SerialNumber] {
```

Comment:
I think this still needs the check for videoCaptureDevice, because it is possible for the streams to switch around, and that a stream attached to the device is still not one that we expect to be stored